R/calculate.R

```diff
@@ -152,3 +162,40 @@ calculate_exposure_by_technology_and_company <- function(asset_type,
 
   return(exposure_by_technology_and_company_filtered)
 }
+
+calculate_terminal_value <- function(data,
+                                     end_year,
+                                     growth_rate,
+                                     discount_rate) {
+  # the calculation follows the formula described in the 2DII paper "Limited
+  # Visibility", available under https://2degrees-investing.org/resource/limited-visibility-the-current-state-of-corporate-disclosure-on-long-term-risks/
+  terminal_value <- data %>%
+    dplyr::filter(.data$year == .env$end_year) %>%
+    # ADO 3112: ideally NPS and SDS (or more generally the columns with baseline
+    # and target scenario names) should also get NA values. This makes no
+    # difference for the rest of the analysis, but would be prettier.
+    dplyr::mutate(
+      year = .env$end_year + 1,
+      net_profits_baseline = .data$net_profits_baseline * (1 + .env$growth_rate),
+      net_profits_ls = .data$net_profits_ls * (1 + .env$growth_rate),
+      discounted_net_profit_baseline = .data$net_profits_baseline /
+        (.env$discount_rate - .env$growth_rate),
+      discounted_net_profit_ls = .data$net_profits_ls /
+        (.env$discount_rate - .env$growth_rate),
+      baseline = NA_real_,
+      scen_to_follow_aligned = NA_real_,
+      late_sudden = NA_real_,
+      scenario_change_aligned = NA_real_,
+
+    )
+
+  data <- data %>%
+    dplyr::bind_rows(terminal_value) %>%
+    dplyr::arrange(
+      .data$investor_name, .data$portfolio_name, .data$id,
+      .data$scenario_geography, .data$company_name, .data$ald_sector,
```

Comment:
As a last remark before I am approving it. Wenn comparing 2040 and 2041 it shows that in some cases info for 2041 is NA, and in some cases it is identical to 2040,  (e.g. proximity to target). It might be wise to revise per column if the way the columns are filled now is really correct and implement this TODO ```  # ADO 3112: ideally NPS and SDS (or more generally the columns with baseline
    # and target scenario names) should also get NA values. This makes no
    # difference for the rest of the analysis, but would be prettier.``` now, before creating confusion in the future

Reply:
@MirijaHa sure, question is how best to get the scenario names without hard coding SDS and NPS again. But more than happy to include

Reply:
please hardcode them now so that the basis is correct and make it a problem for me to be resolved in https://dev.azure.com/2DegreesInvesting/2DegreesInvesting/_boards/board/t/Stress%20Testing%20Developers/Backlog%20items/?workitem=3129

Reply:
no need for a follow up, solved with sym()